File: cpp/PE019.cpp
#include <iostream>
using namespace std;

int main()
{
    int a[] = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
    int i, j, count = 0, daySince = 0;
    for(i = 1901; i <= 2000; i++){
        for(j = 0; j < 12; j++){
            daySince += a[j];
            if((j == 1) && (i % 4) == 0)
                daySince ++;
            if((daySince % 7) == 5)
                count ++;
        }
    }
    cout << count << endl;
    return 0;
}

File: python/PE019.py
#!/usr/bin/env python
def isLeap(n):
    if (n%4 == 0 and n%100 != 0) or n%400 == 0:
        return True
    return False

def main():
    day = [
            [31, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30], 
            [31, 31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30]
            ]
    count = 0
    weekday = 1-31
    for i in range(1900, 2001):
        for j in range(12):
            weekday += day[isLeap(i)][j]
            weekday %= 7
            if weekday == 0 and i != 1900:
                count += 1
    print count

if __name__ == '__main__':
    main()
